sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -57,7 +57,9 @@ class GpuProjectExecMeta(
     val gpuExprs = childExprs.map(_.convertToGpu().asInstanceOf[NamedExpression]).toList
     val gpuChild = childPlans.head.convertIfNeeded()
     if (conf.isProjectAstEnabled) {
-      if (childExprs.forall(_.canThisBeAst)) {
+      if (childExprs.forall(_.canThisBeAst)
+        // cuDF requires return column is fix width
+        && gpuExprs.forall(expr => GpuBatchUtils.isFixedWidth(expr.dataType))) {
```

Comment:
nit, we could cache the result of this so we don't have to re-run it for the warning later?

Reply:
Done, now cache the reuslt.